callstack/react-native-builder-bob
# feat(crnl): store metadata with new projects

### Summary

We need a way to track the project information since we are going to introduce a library upgrade helper. With this PR, when you create a new library using `create-react-native-library`, it will automatically save;

1. Version of `create-react-native-library` used to create the project
2. The library type (native module, native view, etc.)
3. The languages used (Kotlin, objective-c, etc.)

into `create-react-native-library.json` file at the root.

### Test plan


1. Build and run `create-react-native-library`
2. Note down the options you have picked
3. Make sure the `create-react-native-library.json` file has been generated at the root of your library
4. Check the version, library type, and languages fields and make sure they are using correct data.
